src/imbi_automations/models/imbi.py

```diff
@@ -7,16 +7,39 @@
 
 import typing
 
+import pydantic
+
 from . import base
 
 
 class ImbiEnvironment(base.BaseModel):
-    """Imbi environment with metadata."""
+    """Imbi environment with metadata.
+
+    The slug field is auto-generated from the name if not provided,
+    converting to lowercase and replacing spaces with hyphens.
+    """
 
     name: str
+    slug: str | None = None
```

Comment:
You can make slugs required now and remove some complexity unless this is an issue with the on-disk cache.